Languages/IronPython/IronPython.Modules/gc.cs

```diff
@@ -78,9 +78,22 @@ public static class PythonGC {
         }
 
         public static void set_threshold(CodeContext/*!*/ context, params object[] args) {
+            if(args.Length == 0) {
+                throw PythonOps.TypeError("set_threshold() takes at least 1 argument (0 given)");
+            }
+
+            if(args.Length > 3) {
+                throw PythonOps.TypeError("set_threshold() takes at most 3 arguments ({0} given)", args.Length);
+            }
+
+            if(args.Any(x => x is float)) {
```

Comment:
My guess is it's not displaying this particular error message and falling through to your next check.